### PR TITLE
Refactored externalSigner to use Typedefs

### PR DESCRIPTION
**Pull request by rolfyone**
- created a new suite of tests to show a number of different signing bodies.
 - had to do manual testing for external signer because no AT i think
 - I was able to get from phase0 -> bellatrix so this should cover most scenarios.
 - I believe most of the wrappers should be available in the api

partially addresses #8527

## Documentation

- [X] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [X] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: acceptance-tests/src/acceptance-test/java/tech/pegasys/teku/test/acceptance/CapellaRemoteSignerAcceptanceTest.java [hunk 1]
```diff
@@ -39,7 +39,7 @@ public class CapellaRemoteSignerAcceptanceTest extends AcceptanceTestBase {
   private static final URL JWT_FILE = Resources.getResource("auth/ee-jwt-secret.hex");
 
   @Test
-  void denebWithRemoteSigner() throws Exception {
+  void capellaWithRemoteSigner() throws Exception {
     final UInt64 currentTime = new SystemTimeProvider().getTimeInSeconds();
     final int genesisTime =
         currentTime.intValue() + 10; // genesis in 10 seconds to give node time to start
```
Comment (lucassaldanha) [inline acceptance-tests/src/acceptance-test/java/tech/pegasys/teku/test/acceptance/CapellaRemoteSignerAcceptanceTest.java]: Do we want 10 or 30 seconds? :)

File: validator/client/src/integration-test/java/tech/pegasys/teku/validator/client/signer/ExternalSignerAltairIntegrationTest.java [hunk 2]
```diff
@@ -60,33 +62,6 @@ public Spec getSpec() {
     return TestSpecFactory.createMinimalAltair();
   }
 
-  @Test
-  void shouldSignAltairBlock() throws Exception {
-    final BeaconBlock block = dataStructureUtil.randomBeaconBlock(10);
```
Comment (gfukushima) [inline validator/client/src/integration-test/java/tech/pegasys/teku/validator/client/signer/ExternalSignerAltairIntegrationTest.java]: what's the reason we're removing this test?

**Review by gfukushima — APPROVED** [review]
lgtm, just a question about a Integration test that was removed

File: acceptance-tests/src/acceptance-test/java/tech/pegasys/teku/test/acceptance/CapellaRemoteSignerAcceptanceTest.java (hunk 1 above)
Comment (rolfyone) [reply acceptance-tests/src/acceptance-test/java/tech/pegasys/teku/test/acceptance/CapellaRemoteSignerAcceptanceTest.java]: i did this one before the rest, i think we decided 30...

File: validator/client/src/integration-test/java/tech/pegasys/teku/validator/client/signer/ExternalSignerAltairIntegrationTest.java (hunk 2 above)
Comment (rolfyone) [reply validator/client/src/integration-test/java/tech/pegasys/teku/validator/client/signer/ExternalSignerAltairIntegrationTest.java]: it's sort of redundant but i'll see if i can get it to work now...

Comment (rolfyone) [reply validator/client/src/integration-test/java/tech/pegasys/teku/validator/client/signer/ExternalSignerAltairIntegrationTest.java]: ok, it was an actual issue.... I had intended to loop back on this post the integration test changes, I was hoping it'd come in fixed, and it did... but still didnt work, so it's back and now it works.

File: validator/api/src/main/java/tech/pegasys/teku/validator/api/signer/BlockWrapper.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  }
+
+  private SerializableTypeDefinition<BeaconBlockHeader> getMaybeBlockHeaderSchema() {
+    return blockHeader
+        .<SerializableTypeDefinition<BeaconBlockHeader>>map(
+            beaconBlockHeader -> beaconBlockHeader.getSchema().getJsonTypeDefinition())
+        .orElse(null);
+  }
+
+  static class Builder {
```
[the remark this answers is not in the document]
Comment (rolfyone) [reply validator/api/src/main/java/tech/pegasys/teku/validator/api/signer/BlockWrapper.java]: fixed.

**Pull request merged**